src/v/cluster/rm_stm.cc

```diff
@@ -433,6 +467,22 @@ ss::future<tx_errc> rm_stm::abort_tx(
         _mem_state = mem_state{.term = _insync_term};
     }
 
+    auto origin = get_abort_origin(pid, tx_seq);
+    if (origin == abort_origin::past) {
+        // rejecting a delayed abort command to prevent aborting
+        // a wrong transaction
+        co_return tx_errc::timeout;
+    } else if (origin == abort_origin::future) {
+        // impossible situation: before transactional coordinator may issue
+        // abort of the current transaction it should begin it and abort all
+        // previous transactions with the same pid
+        vlog(
+          clusterlog.error,
+          "Rejecting abort because it isn't consistent with "
+          "the current ongoing transaction");
+        co_return tx_errc::timeout;
```

Comment:
I understand the semantics of `tx_errc::timeout` for `abort_origin::past`. Not sure if `timeout` error is the one we should return in case of `abort_origin::future`

Reply:
The general meaning of timeout is that a request has times out and its true status is unknown, it may pass or fail. In the network applications we ought to handle this situation. So when I'm being lazy and don't want to introduce a new error type and a handler I just fall back to timeout since nothing can go bad if I return it in any situation.

I'll fix it to return an explicit error code.